### DISCUSSION TITLE
404 Not found from azd init --template todo-csharp-cosmos-sql

### DISCUSSION THREAD

**Discussion by siegfried01** [Azure/azure-dev#732] [Q&A] [answered]
I'm following this tutorial:

[https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/get-started?tabs=devcontainer%2Cwindows&pivots=programming-language-csharp#initialize-project](url)

I just upgraded azd prior to issuing the command

    azd init --template todo-csharp-cosmos-sql

After it deploys, I go to the azure portal -> click on resource group -> click on web app -> click on browse button -> "404 not found".

Siegfried

**Comment by weikanglim** [comment, marked as answer]
@siegfried01 Can you verify the web app resource you clicked on is prefixed with `app-web` (the front-end web app) and not `app-api` (the backend API server)? The backend `app-api` for C# currently does not have a root endpoint mapped, which is likely why you received a 404. This is being fixed to show a Swagger UI interface definition instead.

**Reply by siegfried01** [reply]
Yes... I believe it was the app-web and not the app-api... I don't believe the portal display for the app-api has a browse button. Also: I checked the logs (log stream) and the error was that the web server was looking for index.html and could not find it. I suspect the compiled C# web app code did not deploy to the web site.

**Reply by weikanglim** [reply]
OK. There should be a browse button for both web apps. Did you run `azd up` instead of `azd init`? Just double checking -- I think the original thread says `azd init` but that is likely just a typo.

Was the `azd up` also successful? Making sure it did not fail the deployment half-way.

Re-running `azd up` is likely the easiest fix -- if it doesn't, then it might be some silent failure happening with the deployment.

**Reply by siegfried01** [reply]
OK, I tried again after deleting the directory contents on my computer and deleting the resource group and purging the key vault on the portal:

    PS C:\Users\shein\Documents\temp\azd todo-csharp-cosmos-sql> azd up --template todo-csharp-cosmos-sql
    Initializing a new project in C:\Users\shein\Documents\temp\azd todo-csharp-cosmos-sql
    
    ? Please enter a new environment name: todo-csharp-cosmos-sql
    
    ? Please enter a new environment name: todo-csharp-cosmos-sql
    ? Please select an Azure location to use: 49. (US) West US 2 (westus2)
    ? Please select an Azure Subscription to use:  1. Azure subscription 1 (acc26051-92a5-4ed1-a226-64a187bc27db)
    
    Infrastructure provisioning plan completed successfully
    Provisioning Azure resources can take some time.
    
    You can view detailed progress in the Azure Portal:
    https://portal.azure.com/#blade/HubsExtension/DeploymentDetailsBlade/overview/id/%2Fsubscriptions%2Facc26051-92a5-4ed1-a226-64a187bc27db%2Fproviders%2FMicrosoft.Resources%2Fdeployments%2Ftodo-csharp-cosmos-sql
    
    Created Resource group: rg-todo-csharp-cosmos-sql
    Created Log Analytics workspace: log-gqb3tgcqh5qys
    Created App Service plan: plan-gqb3tgcqh5qys
    Created Key vault: kv-gqb3tgcqh5qys
    Created Application Insights: appi-gqb3tgcqh5qys
    Created Portal dashboard: dash-gqb3tgcqh5qys
    Created App Service: app-web-gqb3tgcqh5qys
    Created App Service: app-api-gqb3tgcqh5qys
    Created Azure Cosmos DB: cosmos-gqb3tgcqh5qys
    
    Azure resource provisioning completed successfully
    
    Deploying service api...
    Error: deploying service: packaging service api: dotnet publish on project 'C:\Users\shein\Documents\temp\azd todo-csharp-cosmos-sql\src\api' failed: exit code: 1, stdout: MSBuild version 17.3.0-preview-22329-01+77c72dd0f for .NET
    C:\Users\shein\Documents\temp\azd todo-csharp-cosmos-sql\src\api\Todo.Api.sln.metaproj : error MSB4126: The specified solution configuration "Release|x64" is invalid. Please specify a valid solution configuration using the Configuration and Platform properties (e.g. MSBuild.exe Solution.sln /p:Configuration=Debug /p:Platform="Any CPU") or leave those properties blank to use the default solution configuration. [C:\Users\shein\Documents\temp\azd todo-csharp-cosmos-sql\src\api\Todo.Api.sln]
    , stderr: : exit status 1
    PS C:\Users\shein\Documents\temp\azd todo-csharp-cosmos-sql>
